### PR TITLE
add another business functionality changes

### PR DESCRIPTION
**Pull request by chandraAndem**

### CODE REVIEW

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala [hunk 1]
```diff
@@ -107,6 +109,106 @@ class BusinessListCYAControllerSpec extends ControllerBaseSpec with MockIncomeTa
...
+
+        val result: Future[Result] = TestBusinessListCYAController.submit()(FakeRequest()
+          .withFormUrlEncodedBody(AddAnotherBusinessForm.addAnotherBusiness -> YesNoMapping.option_yes))
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.InitialiseController.initialise().url)
+
+      }
+
+      "the connector returns successful json and submit with No option redirect to Initialse controller" in {
```
Comment (jameshforster) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala:128]: Could you rename this test to correctly match the scenario

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala [hunk 2]
```diff
@@ -107,6 +109,106 @@ class BusinessListCYAControllerSpec extends ControllerBaseSpec with MockIncomeTa
 
   }
 
+  "submit" should {
+    "return 303, SEE_OTHER" when {
+      "the connector returns successful json and submit with Yes option redirect to Initialse controller" in {
```
Comment (jameshforster) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala:114]: Could you fix the spelling of Initialise please

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala [hunk 3]
```diff
@@ -107,6 +109,106 @@ class BusinessListCYAControllerSpec extends ControllerBaseSpec with MockIncomeTa
...
+
+    }
+
+    "return (303) redirect to business start date page" when {
+      "no businesses are returned" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments[Seq[SelfEmploymentData]]("Businesses")(Right(None))
+
+        val result: Future[Result] = TestBusinessListCYAController.submit()(FakeRequest()
+          .withFormUrlEncodedBody(AddAnotherBusinessForm.addAnotherBusiness -> YesNoMapping.option_yes))
```
Comment (jameshforster) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala:149]: This test isn't really meaningful unless you set the option to "No". It has no behaviour difference to selecting "Yes" in any circumstance.

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala [hunk 4]
```diff
@@ -107,6 +109,106 @@ class BusinessListCYAControllerSpec extends ControllerBaseSpec with MockIncomeTa
...
+        redirectLocation(result) mustBe Some(routes.InitialiseController.initialise().url)
+      }
+      "no complete businesses are returned" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments[Seq[SelfEmploymentData]]("Businesses")(Right(Some(Seq(
+          businessData.copy(businessName = None)
+        ))))
+
+        val result: Future[Result] = TestBusinessListCYAController.submit()(FakeRequest()
+          .withFormUrlEncodedBody(AddAnotherBusinessForm.addAnotherBusiness -> YesNoMapping.option_yes))
```
Comment (jameshforster) [inline test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala:160]: This test isn't really meaningful unless you set the option to "No". It has no behaviour difference to selecting "Yes" in any circumstance.

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala (hunk 1 above)
Comment (chandraAndem) [reply test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala]: Done

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala (hunk 2 above)
Comment (chandraAndem) [reply test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala]: Done

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala (hunk 3 above)
Comment (chandraAndem) [reply test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala]: That's correct, I was only looking at empty getselfemployments. 
Done.

File: test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala (hunk 4 above)
Comment (chandraAndem) [reply test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala]: Done

**Pull request merged**